web/src/stream_settings_ui.js

```diff
@@ -705,17 +705,20 @@ export function change_state(section, right_side_tab) {
             show_right_section();
         } else {
             toggler.goto("subscribed");
+            stream_edit.empty_right_panel();
         }
         return;
     }
 
     if (section === "all") {
         toggler.goto("all-streams");
+        stream_edit.empty_right_panel();
         return;
     }
 
     if (section === "subscribed") {
         toggler.goto("subscribed");
+        stream_edit.empty_right_panel();
```

Comment:
Have called this here instead of in `callback` function called from `goto` because we are updating the right panel to be empty due to change in hash and not because of toggling between sections and it also helps in avoiding to update hash multiple times.

Reply:
So to be clear -- this means we empty the right panel when navigating by hash, but don't empty the right panel when we switch between the all/subscribed tabs using the tab switcher. I think that makes sense but just want to clarify that's what we want.

Reply:
Yes. We only empty the right panel while toggling between all and subscribed tabs if an unsubscribed stream was opened in the "all" and then I toggle to "subscribed" tab. There is another PR #28481 to fix the hash and title in right panel for such cases.